### PR TITLE
feat(types): export NavigationGuardNext

### PR DESCRIPTION
**Pull request by cexbrayat**
You currently have to type the `next` arg manaully in a navigation guard

### CODE REVIEW

File: test-dts/createRouter.test-d.ts [hunk 1]
```diff
@@ -14,5 +20,12 @@ const router = createRouter({
   scrollBehavior(to, from, savedPosition) {},
 })
 
+export const loggedInGuard: NavigationGuard = (
```
Comment (posva) [inline test-dts/createRouter.test-d.ts:23]: isn't this type redundant? I think the type arguments make more sense with a plain function:


```js
function otherGuard(to: RouteLocationNormalized, from: RouteLocationNormalized, next: NavigationGuardCallback) {}
```

I also think we should rename it to `NavigationGuardNext`, don't do it in the PR to keep it simple but what do you think?

Comment (cexbrayat) [reply test-dts/createRouter.test-d.ts:23]: I think I agree that `NavigationGuard` is not super necessary. `NavigationGuardNext` is probably better indeed 👍 And `RouteLocationNormalized` could be simplified to `RouteLocation`?

Comment (posva) [reply test-dts/createRouter.test-d.ts:23]: I want to but there are a few more types that represent different kinds of route locations. I think I need to do some refactoring when it comes to naming and would definitely welcome some feedback

File: test-dts/createRouter.test-d.ts [hunk 2]
```diff
@@ -14,5 +20,12 @@ const router = createRouter({
...
+  to: RouteLocationNormalized,
+  from: RouteLocationNormalized,
+  next: NavigationGuardCallback
+) => next('/')
+router.beforeEach(loggedInGuard)
```
Comment (posva) [inline test-dts/createRouter.test-d.ts:28]:
```suggestion
export const loggedInGuard: NavigationGuard = (to, from, next) => next('/')
function beforeGuardFn(
  to: RouteLocationNormalized,
  from: RouteLocationNormalized,
  next: NavigationGuardCallback
) {}

router.beforeEach(loggedInGuard)
router.beforeEach(beforeGuardFn)
```

**Pull request merged**